### PR TITLE
Two integrations don't work at the same time

### PR DESCRIPTION
**Pull request by sdvendramini**
|Related issue|Manual Testing|
|---|---|
|https://github.com/wazuh/wazuh/issues/4264|https://github.com/wazuh/wazuh-qa/issues/3375|

## Description
Currently integratord create a tmp file per alert. These changes where made to create a tmp file per alert and integration type (slack, pagerduty ...) as the alert formatting may differ between each integration.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux

- Memory tests for Linux
  - [x] Valgrind (memcheck and descriptor leaks check)
  - [x] AddressSanitizer

### CODE REVIEW

File: src/os_integrator/integrator.c [hunk 1]
```diff
@@ -54,6 +54,9 @@ void OS_IntegratorD(IntegratorConfig **integrator_config)
     /* Connecting to syslog. */
     while(integrator_config[s])
     {
+        #ifdef WAZUH_UNIT_TESTING
```
Comment (TomasTurina) [inline src/os_integrator/integrator.c:57]: Why do you do this? If you don't want to test this code, setting `integrator_config[0]` in the test would be enough.

**Review by TomasTurina — CHANGES_REQUESTED** [review]

File: src/os_integrator/integrator.c (hunk 1 above)
Comment (sdvendramini) [reply src/os_integrator/integrator.c]: Done!

**Review by TomasTurina — APPROVED** [review]

**Review by TomasTurina — APPROVED** [review]

**Review by jmv74211 — APPROVED** [review]
## QA review

- **Type**: Manual testing.
- **Status**: Approved  🟢 
- **Testing issue**: https://github.com/wazuh/wazuh-qa/issues/3375
- **Comments**:  Everything seems to be working properly

**Comment by branchnetconsulting** [comment]
If there are several Slack integration sections with legitimately overlapping criteria, such that more that one of them may be triggered on the same unique alert, will this PR ensure that each call to the Slack integration for that one alert, will use its own unique tmp file?

**Pull request merged**